src/Paprika/Store/StorageFanOutPage.cs

```diff
@@ -107,18 +107,31 @@ public void Destroy(IBatchContext batch, in NibblePath prefix)
         // Destroy the Id entry about it
         Set(prefix, ReadOnlySpan<byte>.Empty, batch);
 
-        // Destroy the account entry
-        // SetAtRoot<FanOutPage>(batch, account, ReadOnlySpan<byte>.Empty, ref Data.StateRoot);
-
-        // Remove the cached
-        batch.IdCache.Remove(prefix.UnsafeAsKeccak);
-        var index = GetIndex(prefix);
-        var addr = Data.Addresses[index];
-        if (addr.IsNull)
+        var map = new SlottedArray(Data.Data);
+        foreach (var item in map.EnumerateAll())
         {
-            // recycleForReuse
+            if (item.Key.Equals(prefix))
```

Comment:
Not equals but `StartsWith`. I think this method is missing. If it is, it can be implemented within `NibblePath` using `FindFirstDifferentNibble` and checking for the length

Reply:
Is the Equal method defined in NibblePath doing it differently? - (https://github.com/NethermindEth/Paprika/blob/8b91e46d65eb296ba96dc2004a847a9504feb68c/src/Paprika/Data/NibblePath.cs#L4)

Reply:
You don't want to equal, you want to destroy also all the merkle nodes that have a prefix.

Reply:
Okay, something like this  - 

    public bool StartsWith(in NibblePath other)
    {
        return FindFirstDifferentNibble(other) > 0;
    }